src/utils.cpp

```diff
@@ -2136,30 +2136,121 @@ void NodeCounter::operator -= (const NodeCounter& o)
     versions -= o.versions;
 }
 
-SyncConfig::SyncConfig(std::string localPath,
+
+CacheableStatus::CacheableStatus(int64_t type, int64_t value)
+    : mType{type}, mValue{value}
+{ }
+
+
+// This should be a const-method but can't be due to the broken Cacheable interface.
+// Do not mutate members in this function! Hence, we forward to a private const-method.
+bool CacheableStatus::serialize(std::string* data)
+{
+    return const_cast<const CacheableStatus*>(this)->serialize(*data);
+}
+
+std::shared_ptr<CacheableStatus> CacheableStatus::unserialize(class MegaClient *client, const std::string& data)
+{
+    int64_t type;
+    int64_t value;
+
+    CacheableReader reader{data};
+    if (!reader.unserializei64(type))
+    {
+        return {};
+    }
+    if (!reader.unserializei64(value))
+    {
+        return {};
+    }
+
+    auto cacheableStatus = std::make_shared<CacheableStatus>(type, value);
+
+    client->loadCacheableStatus(cacheableStatus);
+    return cacheableStatus;
+}
+
+bool CacheableStatus::serialize(std::string& data) const
+{
+    CacheableWriter writer{data};
+    writer.serializei64(mType);
+    writer.serializei64(mValue);
+    return true;
+}
+
+int64_t CacheableStatus::value() const
+{
+    return mValue;
+}
+
+int64_t CacheableStatus::type() const
+{
+    return mType;
+}
+
+void CacheableStatus::setValue(const int64_t &value)
```

Comment:
No const references for primitive datatypes that are smaller than the management of the indirection (typically 64bits), since it actually worsen the performance. I'll change this one.